internal/runtime/hcsv2/nvidia_utils.go

```diff
@@ -0,0 +1,77 @@
+// +build linux
+
+package hcsv2
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/Microsoft/opengcs/internal/storage/pci"
+	oci "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
+const lcowNvidiaMountPath = "/run/nvidia"
+
+// addNvidiaDevicePreHook builds the arguments for nvidia-container-cli and creates the prestart hook
+func addNvidiaDevicePreHook(ctx context.Context, spec *oci.Spec) error {
+	nvidiaToolBinary := "nvidiaPrestartHook"
+	nvidiaToolPath, err := exec.LookPath(nvidiaToolBinary)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find %s for container GPU support", nvidiaToolBinary)
+	}
+
+	var args []string
+	args = append(args, nvidiaToolPath, "--debug=/nvidia-container.log", "--load-kmods", "--no-pivot", "configure", "--ldconfig=/sbin/ldconfig")
+	if capabilities, ok := spec.Annotations["io.microsoft.container.gpu.capabilities"]; ok {
+		caps := strings.Split(capabilities, ",")
+		for _, c := range caps {
+			args = append(args, fmt.Sprintf("--%s", c))
+		}
+	}
+
+	for _, d := range spec.Windows.Devices {
+		switch d.IDType {
+		case "gpu":
+			busLocation, err := pci.FindDeviceBusLocationFromVMBusGUID(ctx, d.ID)
+			if err != nil {
+				return errors.Wrapf(err, "failed to find nvidia gpu bus location")
+			}
+			args = append(args, fmt.Sprintf("--device=%s", busLocation))
+		}
+	}
+
+	rootfsDir := spec.Root.Path
+	args = append(args, "--no-cgroups", "--pid=%v", fmt.Sprintf("%s", rootfsDir))
+
+	if spec.Hooks == nil {
+		spec.Hooks = &oci.Hooks{}
+	}
+
+	nvidiaHook := oci.Hook{
+		Path: nvidiaToolPath,
+		Args: args,
+		Env:  updateEnvWithNvidiaPath(),
+	}
+
+	spec.Hooks.Prestart = append(spec.Hooks.Prestart, nvidiaHook)

```

Comment:
I seem to remember something about `Prestart` being deprecated. Is that the case? Should we add a comment here about it?

Reply:
Its use is going to be deprecated, but the new [hooks described in the runtime spec](https://github.com/opencontainers/runtime-spec/blob/39c287c415bf86fb5b7506528d471db5405f8ca8/config.md#posix-platform-hooks) have not yet been implemented in runc. 

Looks like there is active work and [discussion](https://github.com/opencontainers/runc/pull/1811#issuecomment-582682968) to implement the new hooks, but it's unclear when that will be done. 

I will add a comment to indicate we should update to the new hooks asap.